src/plugins/gs-plugin-eos.c

```diff
@@ -303,15 +301,13 @@ get_app_locale_cache_key (const char *app_name)
 }
 
 static gboolean
-gs_plugin_locale_cached_app_is_best_match (GsPlugin *plugin,
-					   const char *locale_cache_key)
+gs_plugin_app_is_locale_best_match (GsPlugin *plugin,
+				    GsApp *app)
 {
-	GsApp *cached_app = gs_plugin_cache_lookup (plugin, locale_cache_key);
-
-	if (!cached_app)
+	if (!app)
```

Comment:
Can you check `app != NULL` before calling into this function? Or even put the `g_str_has_suffix` there and remove this function